### PR TITLE
Add k3s server object output

### PR DESCRIPTION
**Pull request by dkeightley**
Closes: https://github.com/rancherlabs/support-tools/issues/34

### CODE REVIEW

File: collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh [hunk 1]
```diff
@@ -337,10 +337,14 @@ k3s-rancher() {
 
   if [ -d /var/lib/rancher/k3s/server ]; then
     unset KUBECONFIG
-    k3s kubectl get events --all-namespaces > $TMPDIR/k3s/kubectl/events 2>&1
-    k3s kubectl get endpoints -o wide --all-namespaces > $TMPDIR/k3s/kubectl/endpoints 2>&1
-    k3s kubectl get configmaps --all-namespaces > $TMPDIR/k3s/kubectl/configmaps 2>&1
-    k3s kubectl get namespaces > $TMPDIR/k3s/kubectl/namespaces 2>&1
+    K3S_OBJECTS=(clusterroles clustrerrolebindings crds mutatingwebhookconfigurations namespaces nodes pv validatingwebhookconfigurations)
```
Comment (dnoland1) [inline collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh:340]: Typo, got error:
```
error: the server doesn't have a resource type "clustrerrolebindings"
```

File: collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh [hunk 2]
```diff
@@ -337,10 +337,14 @@ k3s-rancher() {
...
+    K3S_OBJECTS_NAMESPACED=(apiservices configmaps deployments daemonsets endpoints events helmcharts hpa ingress jobs leases pods pvc replicasets roles rolebindings statefulsets)
+    for OBJECT in "${K3S_OBJECTS[@]}"; do
+      k3s kubectl get ${OBJECT} > $TMPDIR/k3s/kubectl/${OBJECT} 2>&1
```
Comment (dnoland1) [inline collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh:343]: Any reason not to use `-o wide`? That will give a few more fields in the output that might be useful.

File: collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh [hunk 3]
```diff
@@ -337,10 +337,14 @@ k3s-rancher() {
 
   if [ -d /var/lib/rancher/k3s/server ]; then
     unset KUBECONFIG
-    k3s kubectl get events --all-namespaces > $TMPDIR/k3s/kubectl/events 2>&1
-    k3s kubectl get endpoints -o wide --all-namespaces > $TMPDIR/k3s/kubectl/endpoints 2>&1
-    k3s kubectl get configmaps --all-namespaces > $TMPDIR/k3s/kubectl/configmaps 2>&1
-    k3s kubectl get namespaces > $TMPDIR/k3s/kubectl/namespaces 2>&1
+    K3S_OBJECTS=(clusterroles clustrerrolebindings crds mutatingwebhookconfigurations namespaces nodes pv validatingwebhookconfigurations)
+    K3S_OBJECTS_NAMESPACED=(apiservices configmaps deployments daemonsets endpoints events helmcharts hpa ingress jobs leases pods pvc replicasets roles rolebindings statefulsets)
```
Comment (dnoland1) [inline collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh:341]: `cronjobs` might be another nice one to have.

**Comment by dnoland1** [comment]
Might also be nice to have output of:
```
k3s check-config
kubectl api-resources
```

**Comment by dkeightley** [comment]
Thanks, I added these suggestions.

I think we have [k3s check-config](https://github.com/rancherlabs/support-tools/blob/a3c04bbfe6a41858663f4b7d33c6e472018a44ba/collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh#L251) covered for server/agents at the moment

**Review by dnoland1 — APPROVED** [review]
LGTM, tested on a k3s node running Ubuntu 20.04.

**Pull request merged**